### ISSUE TITLE
petstore 1.0.18 has wrong scheme!

### ISSUE THREAD

**Issue by lvsecoto** [swagger-api/swagger-petstore#125]
pet.id.type should be integer
```json
"Pet": {
                "required": [
                    "name",
                    "photoUrls"
                ],
                "type": "object",
                "properties": {
                    "id": {
                        "required": [
                            "name",
                            "photoUrls"
                        ],
                        "type": "object",
                        "properties": {
                            "id": {
                                "type": "integer",
                                "format": "int64",
                                "example": 10
                            },
                            "name": {
                                "type": "string",
                                "example": "doggie"
                            },
                            "category": {
                                "$ref": "#/components/schemas/Category"
                            },
                            "photoUrls": {
                                "type": "array",
                                "xml": {
                                    "wrapped": true
                                },
                                "items": {
                                    "type": "string",
                                    "xml": {
                                        "name": "photoUrl"
                                    }
                                }
                            },
                            "tags": {
                                "type": "array",
                                "xml": {
                                    "wrapped": true
                                },
                                "items": {
                                    "$ref": "#/components/schemas/Tag"
                                }
                            },
                            "status": {
                                "type": "string",
                                "description": "pet status in the store",
                                "enum": [
                                    "available",
                                    "pending",
                                    "sold"
                                ]
                            }
                        },
                        "xml": {
                            "name": "pet"
                        }
                    },
                    "name": {
                        "required": [
                            "name",
                            "photoUrls"
                        ],
                        "type": "object",
                        "properties": {
                            "id": {
                                "type": "integer",
                                "format": "int64",
                                "example": 10
                            },
                            "name": {
                                "type": "string",
                                "example": "doggie"
                            },
                            "category": {
                                "$ref": "#/components/schemas/Category"
                            },
                            "photoUrls": {
                                "type": "array",
                                "xml": {
                                    "wrapped": true
                                },
                                "items": {
                                    "type": "string",
                                    "xml": {
                                        "name": "photoUrl"
                                    }
                                }
                            },
                            "tags": {
                                "type": "array",
                                "xml": {
                                    "wrapped": true
                                },
                                "items": {
                                    "$ref": "#/components/schemas/Tag"
                                }
                            },
                            "status": {
                                "type": "string",
                                "description": "pet status in the store",
                                "enum": [
                                    "available",
                                    "pending",
                                    "sold"
                                ]
                            }
                        },
                        "xml": {
                            "name": "pet"
                        }
                    },
                    "category": {
                        "required": [
                            "name",
                            "photoUrls"
                        ],
                        "type": "object",
                        "properties": {
                            "id": {
                                "type": "integer",
                                "format": "int64",
                                "example": 10
                            },
                            "name": {
                                "type": "string",
                                "example": "doggie"
                            },
                            "category": {
                                "$ref": "#/components/schemas/Category"
                            },
                            "photoUrls": {
                                "type": "array",
                                "xml": {
                                    "wrapped": true
                                },
                                "items": {
                                    "type": "string",
                                    "xml": {
                                        "name": "photoUrl"
                                    }
                                }
                            },
                            "tags": {
                                "type": "array",
                                "xml": {
                                    "wrapped": true
                                },
                                "items": {
                                    "$ref": "#/components/schemas/Tag"
                                }
                            },
                            "status": {
                                "type": "string",
                                "description": "pet status in the store",
                                "enum": [
                                    "available",
                                    "pending",
                                    "sold"
                                ]
                            }
                        },
                        "xml": {
                            "name": "pet"
                        }
                    },
                    "photoUrls": {
                        "required": [
                            "name",
                            "photoUrls"
                        ],
                        "type": "object",
                        "properties": {
                            "id": {
                                "type": "integer",
                                "format": "int64",
                                "example": 10
                            },
                            "name": {
                                "type": "string",
                                "example": "doggie"
                            },
                            "category": {
                                "$ref": "#/components/schemas/Category"
                            },
                            "photoUrls": {
                                "type": "array",
                                "xml": {
                                    "wrapped": true
                                },
                                "items": {
                                    "type": "string",
                                    "xml": {
                                        "name": "photoUrl"
                                    }
                                }
                            },
                            "tags": {
                                "type": "array",
                                "xml": {
                                    "wrapped": true
                                },
                                "items": {
                                    "$ref": "#/components/schemas/Tag"
                                }
                            },
                            "status": {
                                "type": "string",
                                "description": "pet status in the store",
                                "enum": [
                                    "available",
                                    "pending",
                                    "sold"
                                ]
                            }
                        },
                        "xml": {
                            "name": "pet"
                        }
                    },
                    "tags": {
                        "required": [
                            "name",
                            "photoUrls"
                        ],
                        "type": "object",
                        "properties": {
                            "id": {
                                "type": "integer",
                                "format": "int64",
                                "example": 10
                            },
                            "name": {
                                "type": "string",
                                "example": "doggie"
                            },
                            "category": {
                                "$ref": "#/components/schemas/Category"
                            },
                            "photoUrls": {
                                "type": "array",
                                "xml": {
                                    "wrapped": true
                                },
                                "items": {
                                    "type": "string",
                                    "xml": {
                                        "name": "photoUrl"
                                    }
                                }
                            },
                            "tags": {
                                "type": "array",
                                "xml": {
                                    "wrapped": true
                                },
                                "items": {
                                    "$ref": "#/components/schemas/Tag"
                                }
                            },
                            "status": {
                                "type": "string",
                                "description": "pet status in the store",
                                "enum": [
                                    "available",
                                    "pending",
                                    "sold"
                                ]
                            }
                        },
                        "xml": {
                            "name": "pet"
                        }
                    },
                    "status": {
                        "required": [
                            "name",
                            "photoUrls"
                        ],
                        "type": "object",
                        "properties": {
                            "id": {
                                "type": "integer",
                                "format": "int64",
                                "example": 10
                            },
                            "name": {
                                "type": "string",
                                "example": "doggie"
                            },
                            "category": {
                                "$ref": "#/components/schemas/Category"
                            },
                            "photoUrls": {
                                "type": "array",
                                "xml": {
                                    "wrapped": true
                                },
                                "items": {
                                    "type": "string",
                                    "xml": {
                                        "name": "photoUrl"
                                    }
                                }
                            },
                            "tags": {
                                "type": "array",
                                "xml": {
                                    "wrapped": true
                                },
                                "items": {
                                    "$ref": "#/components/schemas/Tag"
                                }
                            },
                            "status": {
                                "type": "string",
                                "description": "pet status in the store",
                                "enum": [
                                    "available",
                                    "pending",
                                    "sold"
                                ]
                            }
                        },
                        "xml": {
                            "name": "pet"
                        }
                    }
                },
                "xml": {
                    "name": "pet"
                }
            },
```

**Comment by geoffw** [comment]
The yaml is correct but something from Commits on Mar 8, 2024 has broken the conversion to json and you can see this on the site https://petstore3.swagger.io/#/pet/findPetsByTags if you expand the response schema the datatypes are all wrong.

**Comment by lvsecoto** [comment]
> The yaml is correct but something from Commits on Mar 8, 2024 has broken the conversion to json and you can see this on the site https://petstore3.swagger.io/#/pet/findPetsByTags if you expand the response schema the datatypes are all wrong.

Yes! I also found this problem. Did the Open API json scheme changed?

**Comment by jasonHzq** [comment]
Any progress on this?

**Comment by char0n** [comment]
Yes, we're already looking into this

**Comment by micryc** [comment]
Fixed by #129
